### PR TITLE
Allow same group name within different parent groups

### PR DESCRIPTION
**Pull request by mitiao**
see: https://progress.opensuse.org/issues/17846
feedback and advices please

### CODE REVIEW

**Comment by coolo** [comment]
please add a test case to verify we can have now groups with different parents but same name.

**Comment by Martchus** [comment]
This will likely work, as far as I know the name is really just used for display purposes. But yes, you should provide a test. Just a few more lines in the tests we already have, eg. `t/ui/14-dashboard-parents.t`. You can use the `schema_hook` function to add required fixtures.

**Comment by mitiao** [comment]
@Martchus thanks for your advice.
I am working on the test with a chromedriver issue.
Run whatever test with selenium, for example by `prove -l -v t/ui/14-dashboard-parents.t` and log_chromedriver gave the error:
`[61.063][INFO]: RESPONSE InitSession unknown error: Chrome failed to start: exited abnormally`
any idea?

**Comment by Martchus** [comment]
You could use PhantomJS instead of Chromium. Just ensure the `SELENIUM_CHROME` environment variable is *not* set and it should use PhantomJS.

Note that I'm using Chromium and not Chrome. As far as I remember, I had to install the driver through a separate package under Tumbleweed. So maybe just a package missing?

**Comment by coolo** [comment]
phantomjs support is dead - as IS SELENIUM_CHROME

**Comment by Martchus** [comment]
I'm sometimes still using it and it works (beside on particular line in one particular test). Beside, things might get better when next PhantomJS version is released. It will use Qt WebKit ng/revived (or whatever they want to call it now).

File: lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm [hunk 1]
```diff
@@ -93,12 +93,26 @@ sub list {
     $self->render(json => \@results);
 }
 
+# check existing job group on top level to prevent create/update duplicate
+sub check_top_level_group {
+    my ($self) = @_;
+
+    return 0 if $self->is_parent;
+    my $properties = $self->load_properties;
+    return $self->resultset->search({name => $properties->{'name'}, parent_id => undef});
```
Comment (coolo) [inline lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: JobGroup.pm: Hash key with quotes at line 102, column 60.  Avoid useless quotes.  (Severity: 5)

**Comment by coolo** [comment]
the test cases are problematic at the moment, but the ui tests passed, so we should be fine

**Pull request merged**